com.wamas.ide.launching/src/com/wamas/ide/launching/validation/LcDslValidator.xtend

```diff
@@ -366,12 +367,20 @@ class LcDslValidator extends AbstractLcDslValidator {
 		val type = javaProject.findType(className)
 		if (type === null) {
 			error("Test class " + className + " must point to an existing class in project " + javaProject, cfg.test, LC.testConfig_Class)
-		} else if (!Flags.isPublic(type.flags)) {
-			error("Test class " + className + " must be public.", cfg.test, LC.testConfig_Class)
-		} else if (type.methods.stream().filter([method | Flags.isPublic(method.flags)]).flatMap([method | method.annotations.stream]).noneMatch([annotation | TEST_ANNOTATION_NAMES.contains(annotation.elementName)])) {
-			error("Test class " + className + " must at least have one method annotated with one annotation out of " + TEST_ANNOTATION_NAMES, cfg.test, LC.testConfig_Class)
+		} else if (Flags.isPrivate(type.flags)) {
+			error("Test class " + className + " must not be private.", cfg.test, LC.testConfig_Class)
+		} else if (!hasTestMethod(type) && !hasNestedTest(type)) {
+			error("Test class " + className + " must at least have one method annotated with one annotation out of " + TEST_ANNOTATION_NAMES + " or must contain a nested test.", cfg.test, LC.testConfig_Class)
```

Comment:
I think the point in the example in #50 is not that there **is** a nested test class, but that a (nested) test class is valid in the eyes of the validator if it inherits from a class with test methods.

Reply:
To me that's way to much verification. From my point of view, the lcdsl validation should care about whether this is a test class or not. To me it's perfectly fine if the test gets launched, and only JUnit itself complains about additional things. It's impossible to verify everything that JUnit will verify, so spending effort on more validations seem not beneficial to me.
I currently only care about the validation not blocking any launches because of false positives.

Reply:
One issue with the state of this PR is that nested test classes themselves cannot be run if they do not contain their own test methods or in turn a nested test class. This has been addressed in #72. 

The validation is still not entirely correct, however, because a class can be a test class if it does not fulfil any of conditions in the validator but inherits from another test class (with test methods). So another approach could be to reduce the validation to a basic check for class visibility and let JUnit eventually decide whether it is okay to run it or not. Similar to what you suggested.